File: backend/vehicle-service-backend/src/main/java/com/vsc/vehicle_service_backend/service/SparePartIncomeService.java
package com.vsc.vehicle_service_backend.service;

import com.vsc.vehicle_service_backend.dto.SparePartIncomeRequest;
import com.vsc.vehicle_service_backend.dto.SparePartIncomeResponse;
import java.util.List;
import java.util.Map;

public interface SparePartIncomeService {
    List<SparePartIncomeResponse> getAllIncomes();
    SparePartIncomeResponse getIncomeById(Long id);
    List<SparePartIncomeResponse> getPendingIncomes();
    SparePartIncomeResponse createIncome(SparePartIncomeRequest request);
    SparePartIncomeResponse receiveIncome(Long id);
    SparePartIncomeResponse cancelIncome(Long id);
    void deleteIncome(Long id);
    Map<String, Object> getChartDataByCategory(Long categoryId);
}

File: frontend/src/services/sparePartIncomeService.js
import { api } from './api';
const sparePartIncomeService = {
  // Get all income records - FIXED: removed duplicate /api/v1
  getAllIncomes: () => api.get('/api/v1/spare-part-incomes'),

  // Get income by ID
  getIncomeById: (id) => api.get(`/spare-part-incomes/${id}`),

  // Get pending incomes
  getPendingIncomes: () => api.get('/api/v1/spare-part-incomes/pending'),

  // Create new income
  createIncome: (incomeData) => api.post('/spare-part-incomes', incomeData),

  // Receive income
  receiveIncome: (id) => api.put(`/spare-part-incomes/${id}/receive`),

  // Cancel income
  cancelIncome: (id) => api.put(`/spare-part-incomes/${id}/cancel`),

  // Delete income
  deleteIncome: (id) => api.delete(`/spare-part-incomes/${id}`),

getChartData(categoryId) {
    // Updated to use proper endpoint
    return api.get(`/spare-part-incomes/chart?categoryId=${categoryId}`);
  }
//  // Get chart data by category
//  getChartData: (categoryId) => api.get(`/spare-part-incomes/chart-data/${categoryId}`),
};

export default sparePartIncomeService;